### ISSUE TITLE
Debian support via static executable or deb package

### ISSUE THREAD

**Issue by eliliam** [helix-editor/helix#6824]
There are many users out there who use Debian for both their personal environments as well as in production, I being one of them. While there is a PPA for Ubuntu, this does not seem to work with Debian. I have tried the generic Linux version from the releases page, but am met with errors regarding GLIBC version mismatches. I have experienced this on Debian 11 and the only way that I can seem to get it to work is via building from source. However, building from source is not ideal, especially for smaller environments with minimal compute power and disk space, especially when the rust toolchain is also not available.

![image](https://user-images.githubusercontent.com/14301439/233209623-3d06ca03-f777-46f2-8cdc-eb5d8451e386.png)

My suggestion would be to statically link the executable so that it can run on these systems, or to provide an official deb package that can be used on both Debian as well as Ubuntu.

**Comment by the-mikedavis** [comment]
We don't like to maintain many 'official' packages since there are so many package managers and formats and supporting many is a burden. Also I don't think any of the maintainers use debian frequently or package for it. Like [the other packages](https://repology.org/project/helix/versions) though, it would be great if someone familiar with debian packaging could package Helix

**Comment by bitbegin** [comment]
https://github.com/helix-editor/helix/discussions/6528

**Comment by yunfan** [comment]
@the-mikedavis  then what about the other suggestion on providing static linked executable? it looks like could reduce much the work for providing different linux distrio

**Comment by tgross35** [comment]
I think it is possible for this to be accepted into Debian repos, which should work for both Debian and Ubuntu without a ppa. I think that typically once you produce a `.deb` package, you just need a sponsor who can help get it entered into the bugtracker.

Links to some packages that have recently done the same:

- bat: https://github.com/sharkdp/bat/issues/323#issuecomment-521656639
- delta: https://github.com/dandavison/delta/issues/26

**Comment by tgross35** [comment]
A better tutorial here: https://salsa.debian.org/rust-team/debcargo-conf/blob/master/README.rst

**Comment by pascalkuthe** [comment]
yeah, that would be the ideal, but it requires somebody to volunteer to put the effort into this. As @the-mikedavis mentioned, we don't maintain the distro packages ourselves so somebody else needs to step up to do that (just like for any other distro)

**Comment by tgross35** [comment]
Of course; it would require both somebody to do the changes in this repo and a sponsor on the debian side. I'm just trying to collect some context in case anyone wants to have a go at it

**Comment by pascalkuthe** [comment]
>  do the changes in this repo

I meant that we don't usually accept packaging files in this repo. Ideally, packaging should happen without our involvement (same as for other distros)

**Comment by tgross35** [comment]
Ah; I did not realize that was the case. Do you know off the top of your head who maintains the PPA? (I can figure it out if not). I wonder if the PPA's packaging would be accepted into debian stable with a sponsor.

**Comment by pascalkuthe** [comment]
No, I don't know that I am afraid, I think the PPA also had some issues (or was being abandoned?) but I might be mixing that up it's hard to keep track of all the distro packages. I am also pretty sure that the PPA downloads the official release from github instead of building from source which is a no go for an official package so you would need to start from scratch most likely.

**Comment by tgross35** [comment]
Looks like the context is here https://github.com/helix-editor/helix/issues/3640 and the packaging is at https://github.com/maveonair/helix-ppa. It seems like the rustc version available is prohibitive, but I thought there may be ways to get around that.

@sylvestre I think you maintain Rust packaging for Debian - is there any way Helix could make it into the Debian repos if the MSRV is greater than bookworm's `rustc`?

**Comment by pascalkuthe** [comment]
rustc on debain unstable is new enough( (rustc 1.66) for helix 23.5 and likely the next release too. New packages usually don't get introduced to stable retroactively anyway afaik

**Comment by maveonair** [comment]
> No, I don't know that I am afraid, I think the PPA also had some issues (or was being abandoned?) but I might be mixing that up it's hard to keep track of all the distro packages. I am also pretty sure that the PPA downloads the official release from github instead of building from source which is a no go for an official package so you would need to start from scratch most likely.

No, it's not abandoned, but I had some problems with the build process that were solved thanks to the support of the Helix community. So the PPA is fine, but Debian 12 ships an earlier version of Rustc that cannot be used at the moment to build Helix under Debian 12.

There might be a way to ship a newer version of Rustc via the Debian Backports repository, but I'm not sure if this is the preferred way.

**Comment by pascalkuthe** [comment]
Good to hear the Ubuntu PPA si still maintained.

I think getting into Debian unstable would be a good start. When the next Debian release comes around helix would then be included there

**Comment by FedericoCeratto** [comment]
The progress at building an official Debian package is being tracked at https://bugs.debian.org/1024683

**Comment by tgross35** [comment]
FYI for anyone looking to help - the main TODO is getting a version of all dependencies into [debcargo](https://salsa.debian.org/rust-team/debcargo-conf/-/tree/master). A pretty recent list of the missing packages is here https://salsa.debian.org/debian/hx/-/blob/debian/latest/debian/TODO, mostly various `gix-*` crates
